modules/library/xml/src/main/java/org/geotools/xml/resolver/SchemaCache.java

```diff
@@ -219,48 +217,38 @@ static byte[] download(URI location) {
         return download(location, DEFAULT_DOWNLOAD_BLOCK_SIZE);
     }
 
-    static byte[] download(URI location, int blockSize) {
-        return download(location, blockSize, 0);
-    }
-
     /**
      * Retrieve the contents of a remote URL.
      *
      * @param location and absolute http/https URL.
      * @param blockSize download block size
-     * @param redirectionCount the number of redirection attempts already performed while
-     *     downloading
      * @return the bytes contained by the resource, or null if it could not be downloaded
      */
-    static byte[] download(URI location, int blockSize, int redirectionCount) {
+    static byte[] download(URI location, int blockSize) {
         try {
             URL url = location.toURL();
             String protocol = url.getProtocol();
             if (protocol == null || !(protocol.equals("http") || protocol.equals("https"))) {
                 LOGGER.warning("Unexpected download URL protocol: " + protocol);
                 return null;
             }
-            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-            connection.setConnectTimeout(downloadTimeout);
-            connection.setReadTimeout(downloadTimeout);
-            connection.setUseCaches(false);
-            connection.connect();
-            int responseCode = connection.getResponseCode();
-            if (responseCode != HttpURLConnection.HTTP_OK) {
-                if (hasRedirect(responseCode))
-                    return followRedirect(connection, blockSize, redirectionCount);
-                LOGGER.warning(
-                        String.format(
-                                "Unexpected response \"%d %s\" while downloading %s",
-                                connection.getResponseCode(),
-                                connection.getResponseMessage(),
-                                location.toString()));
+            HTTPClient httpClient = HTTPClientFinder.createClient();
+            httpClient.setConnectTimeout(downloadTimeout);
+            httpClient.setReadTimeout(downloadTimeout);
+            HTTPResponse httpResponse;
+            try {
+                httpResponse = httpClient.get(url);
+            } catch (IOException e) {
+                // don't throw an exception if status code >= 400
```

Comment:
The previous code was logging something when an unexpected response was received:

```
LOGGER.warning(
                        String.format(
                                "Unexpected response \"%d %s\" while downloading %s",
                                connection.getResponseCode(),
                                connection.getResponseMessage(),
                                location.toString()));
```